hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/OptUtils.java

```diff
@@ -482,40 +478,6 @@ public static RexNode inlineExpression(List<RexNode> inlinedExpressions, RexNode
             public RexNode visitInputRef(RexInputRef inputRef) {
                 return inlinedExpressions.get(inputRef.getIndex());
             }
-
-            @Override
-            public RexNode visitLocalRef(RexLocalRef localRef) {
-                return localRef;
-            }
-
-            @Override
-            public RexNode visitCall(RexCall call) {
-                List<RexNode> newOperands = new ArrayList<>(call.getOperands().size());
-                for (RexNode operand : call.operands) {
-                    newOperands.add(operand.accept(this));
-                }
-                return call.clone(call.type, newOperands);
-            }
-
-            @Override
-            public RexNode visitDynamicParam(RexDynamicParam dynamicParam) {
-                return dynamicParam;
-            }
-
-            @Override
-            public RexNode visitFieldAccess(RexFieldAccess fieldAccess) {
```

Comment:
This method on superclass has a bit different semantic

```java
  @Override public RexNode visitFieldAccess(RexFieldAccess fieldAccess) {
    RexNode before = fieldAccess.getReferenceExpr();
    RexNode after = before.accept(this);

    if (before == after) {
      return fieldAccess;
    } else {
      return new RexFieldAccess(
          after,
          fieldAccess.getField());
    }
  }
```

original code throws ```RuntimeException```, the code of superclass does not. Won't this change break anything?

Reply:
In this case the original code was actually broken. We hit the issue with @ivanthescientist with nested fields. In master code, we currently never use `RexFieldAccess`, so it wasn't visible.